### ISSUE TITLE
CSS3 selectors (::selection) result in malformed CSS

### ISSUE THREAD

**Issue by optimalisatie** [pocketjoso/penthouse#36]
CSS3 selectors are causing malformed CSS.

The following source code (part of **all.css**)

```css
::-moz-selection {
  color: #fff;
  background-color: #FE1486
}
::selection {
  color: #fff;
  background-color: #FE1486
}
```

Becomes the following after [grunt-penthouse](https://github.com/pocketjoso/penthouse):

```css
::-moz-selection {
  color: #fff;
  background-color: #FE1486
}
::selection {
  cul{
    list-style: none
  }
  .clear {
    clear: both!important
  }
  footer,header,nav {
    display: block
  }
  :focus {
    outline: 0
  }
  input {
    border: 1px solid #d8d5d5;
    background: #fff;
    padding: 5px
  }
}
```

The element **cul** is unknown (probably malformed code). When the CSS code is minified via cssmin it is further malformed. The bracket of **::selection {** does not have an ending anymore, causing a large part of the CSS file to be removed.

**Comment by pocketjoso** [comment]
Thanks for reporting this bug - will fix it as soon as I have a chance.

**Comment by optimalisatie** [comment]
Thanks a lot for the great plugin! I am looking forward to the update!

**Comment by pocketjoso** [comment]
I can't reproduce your issue - just running the css you pasted here in `penthouse` generates the expected CSS. Please check what css is needed to cause the issue to appear, and let me know. Thanks

**Comment by pocketjoso** [comment]
@optimalisatie Any update? Need to be able to reproduce before I can fix the issue.

**Comment by optimalisatie** [comment]
Hi! I've managed to work around the problem by beautifying the full CSS code via [grunt-cssbeautifier](https://github.com/douzi8/grunt-cssbeautifier).

The full CSS file that was used is:

Minified via ``cssmin`` (causing the problem): https://dev.dermarolling.nl/wp-content/themes/blanco/css/full.raw.css
Beautified: https://dev.dermarolling.nl/wp-content/themes/blanco/css/full.css

The used penthouse conifguration is the following:

```json
penthouse: {
    frontpage : {
        outfile : 'css/abovethefold/frontpage.css',
        css : 'css/full.css',
        url : 'https://dev.dermarolling.nl/',
        width : 1600,
        height : 1200
    },
    product: {
        outfile : 'css/abovethefold/product.css',
        css : 'css/full.css',
        url : 'https://dev.dermarolling.nl/product/titanium-dermaroller/',
        width : 1600,
        height : 1200
    },
    category: {
        outfile : 'css/abovethefold/category.css',
        css : 'css/full.css',
        url : 'https://dev.dermarolling.nl/product-categorie/dermarollers/',
        width : 1600,
        height : 1200
    }
}
```

The resulting files were minified into `inline.min.css` using ``cssmin`` and contained the described malformed CSS code.

The workarround continues as following:

The files are concatenated into 1 file using [grunt-contrib-concat](https://github.com/gruntjs/grunt-contrib-concat):

```json
concat: {
    options: {
        separator: ';',
    },
    dist: {
        src: [
            'css/abovethefold/*.css'
        ],
        dest: 'css/inline.raw.css',
    },
},
```

The concatenated ``penthouse`` CSS is minified via ``cssmin``.

```json
inline_raw: {
    options: {
        banner: '<%= meta.banner %>\n',
        keepSpecialComments: false,
        advanced: true,
        aggressiveMerging: true
    },
    files: {
        'css/inline.raw.css': [
            'css/inline.raw.css'
        ]
    }
}
```

The minified ``penthouse`` CSS is further optimized by splitting the images into a seperate file that is loaded asynchronously using an asynchronous CSS script that is advised by Google (https://developers.google.com/speed/docs/insights/OptimizeCSSDelivery)

```json
split_images: {
    inline: {
        options: {
            output: 'css/inline.images.css'
        },
        files: {
            'css/inline.raw.css': ['css/inline.raw.css']
        }
    }
}
```

```javascript
<script>
var cb = function() {
var l = document.createElement('link'); l.rel = 'stylesheet';
l.href = 'small.css';
var h = document.getElementsByTagName('head')[0]; h.parentNode.insertBefore(l, h);
};
var raf = requestAnimationFrame || mozRequestAnimationFrame ||
  webkitRequestAnimationFrame || msRequestAnimationFrame;
if (raf) raf(cb);
else window.addEventListener('load', cb);
</script>
```

The stripped CSS is then minified again via ``cssmin``.

**Comment by pocketjoso** [comment]
Thanks for the details, I can see the issue now. Will work on a fix.

**Comment by pocketjoso** [comment]
Okay - should be fixed with the latest commit to master. Make sure your `grunt-penthouse` installation pulls the latest penthouse - `0.2.52`.

**Comment by pocketjoso** [comment]
Now properly resolved via #38.
